lib/web_ui/lib/src/engine/html/shaders/shader.dart

```diff
@@ -218,30 +350,95 @@ class GradientRadial extends EngineGradient {
+    _OffScreenCanvas offScreenCanvas =
+    _OffScreenCanvas(widthInPixels, heightInPixels);
+    _GlContext gl = _OffScreenCanvas.supported
+        ? _GlContext.fromOffscreenCanvas(offScreenCanvas._canvas!)
+        : _GlContext.fromCanvas(offScreenCanvas._glCanvas!,
+        webGLVersion == WebGLVersion.webgl1);
+    gl.setViewportSize(widthInPixels, heightInPixels);
+
+    NormalizedGradient normalizedGradient = NormalizedGradient(
+        colors, stops: colorStops);
+
+    _GlProgram glProgram = gl.useAndCacheProgram(
+        _WebGlRenderer.writeBaseVertexShader(),
+        _createRadialFragmentShader(normalizedGradient, tileMode))!;
+
+    Object tileOffset = gl.getUniformLocation(glProgram.program, 'u_tile_offset');
+    double centerX = (center.dx - shaderBounds.left) / (shaderBounds.width);
+    double centerY = (center.dy - shaderBounds.top) / (shaderBounds.height);
+    gl.setUniform2f(tileOffset,
+        2 * (shaderBounds.width * (centerX - 0.5)),
+        2 * (shaderBounds.height * (centerY - 0.5)));
+    Object radiusUniform = gl.getUniformLocation(glProgram.program, 'u_radius');
+    gl.setUniform1f(radiusUniform, radius);
+    normalizedGradient.setupUniforms(gl, glProgram);
+    Object gradientMatrix = gl.getUniformLocation(
+        glProgram.program, 'm_gradient');
+    gl.setUniformMatrix4fv(gradientMatrix, false, matrix4 == null ? Matrix4.identity().storage : matrix4!);
+
+    Object? imageBitmap = _glRenderer!.drawRect(ui.Rect.fromLTWH(0, 0, shaderBounds.width, shaderBounds.height),
+        gl, glProgram, normalizedGradient, widthInPixels, heightInPixels);
+
+    return ctx!.createPattern(imageBitmap!, 'no-repeat')!;
+  }
+
+  String _createRadialFragmentShader(NormalizedGradient gradient,
+      ui.TileMode tileMode) {
+    ShaderBuilder builder = ShaderBuilder.fragment(webGLVersion);
+    builder.floatPrecision = ShaderPrecision.kMedium;
+    builder.addIn(ShaderType.kVec4, name: 'v_color');
+    builder.addUniform(ShaderType.kVec2, name: 'u_resolution');
+    builder.addUniform(ShaderType.kVec2, name: 'u_tile_offset');
+    builder.addUniform(ShaderType.kFloat, name: 'u_radius');
+    builder.addUniform(ShaderType.kMat4, name: 'm_gradient');
+    ShaderDeclaration fragColor = builder.fragmentColor;
+    ShaderMethod method = builder.addMethod('main');
+    // Sweep gradient
+    method.addStatement(
+        'vec2 center = 0.5 * (u_resolution + u_tile_offset);');
+    method.addStatement(
+        'vec4 localCoord = vec4(gl_FragCoord.x - center.x, center.y - gl_FragCoord.y, 0, 1) * m_gradient;');
+    method.addStatement(
+        'float dist = length(localCoord);');
+    method.addStatement(''
+        'float st = abs(dist / u_radius);');
+    final String probeName =
+      _writeSharedGradientShader(builder, method, gradient, tileMode);
+    method.addStatement('${fragColor.name} = ${probeName} * scale + bias;');
+    String shader = builder.build();
```

Comment:
nit: `shader` variable unnecessary, can `return builder.build();`

Reply:
You may want to add more shader code in the future after you calculate fragColor.